applications/adv_networking_bench/cpp/testing/adv_networking_test_harness.py

```diff
@@ -0,0 +1,193 @@
+    logger = logging.getLogger(__name__)
+    logger.debug(f"Validating benchmark results")
+
+    success = True
+    # Check for errors in the log
+    if "[error]" in log:
+        logger.error("Errors found in benchmark output")
+        success = False
+
+    # Check for missed packets
+    transmit_packets = int(re.search(r'Port 0.*?Transmit packets:\s+(\d+)', log, re.DOTALL).group(1))
+    received_packets = int(re.search(r'Port 1.*Received packets:\s+(\d+)', log, re.DOTALL).group(1))
+    received_bytes = int(re.search(r'Port 1.*Received bytes:\s+(\d+)', log, re.DOTALL).group(1))
+    missed_packets = int(re.search(r'Port 1.*Missed packets:\s+(\d+)', log, re.DOTALL).group(1))
+    errored_packets = int(re.search(r'Port 1.*Errored packets:\s+(\d+)', log, re.DOTALL).group(1))
+    exec_time = float(re.search(r'TOTAL EXECUTION TIME OF SCHEDULER : (\d+\.\d+) ms', log).group(1))
+
+    logger.debug("Transmit packets:", transmit_packets)
+    logger.debug("Received packets:", received_packets)
+    logger.debug("Received bytes:", received_bytes)
+    logger.debug("Missed packets:", missed_packets)
+    logger.debug("Errored packets:", errored_packets)
+    logger.debug("Exec time:", exec_time)
+
+    # Check for missed packets
+    missed_packets_ratio = (missed_packets / transmit_packets) * 100
+    if missed_packets_ratio > missed_packets_threshold:
+        logger.error(f"Missed packets: {missed_packets} ({missed_packets_ratio:.2f}%, over {missed_packets_threshold}% threshold ❌)")
+        success = False
+    else:
+        logger.info(f"Missed packets: {missed_packets} ({missed_packets_ratio:.2f}%, below {missed_packets_threshold}% threshold ✅)")
+
+
+    # Check for errored packets
+    errored_packets_ratio = (errored_packets / transmit_packets) * 100
+    if errored_packets_ratio > error_packets_threshold:
+        logger.error(f"Errored packets: {errored_packets} ({errored_packets_ratio:.2f}%, over {error_packets_threshold}% threshold ❌)")
+        success = False
+    else:
+        logger.info(f"Errored packets: {errored_packets} ({errored_packets_ratio:.2f}%, below {error_packets_threshold}% threshold ✅)")
+
+    # Calculate average throughput
+    avg_throughput = (received_bytes * 8) / (exec_time / 1000) / 1e9
+    if avg_throughput < avg_throughput_threshold:
+        logger.error(f"Average throughput: {avg_throughput:.2f} Gbps (below {avg_throughput_threshold} Gbps threshold ❌)")
+        success = False
+    else:
+        logger.info(f"Average throughput: {avg_throughput:.2f} Gbps (above {avg_throughput_threshold} Gbps threshold ✅)")
+
+    return success
+
+
+def main():
+    # Set up logging
+    logging.basicConfig(level=logging.INFO, format='%(asctime)s %(levelname)s: %(message)s')
+
+    # Parse command-line arguments
+    parser = argparse.ArgumentParser(description='Run a bash command')
+    parser.add_argument('command', help='The bash command to run')
+    parser.add_argument('--cwd', help='The working directory')
```

Comment:
Does this add value? Why not execute the script directly from the working directory?

Reply:
To be removed (per offline discussion)